core/chains/evm/chain.go

```diff
@@ -54,7 +55,7 @@ type chain struct {
 	logBroadcaster  log.Broadcaster
 	logPoller       logpoller.LogPoller
 	balanceMonitor  monitor.BalanceMonitor
-	keyStore        keystore.Eth
+	keyStore        txmgr.KeyStore[common.Address, *big.Int, gethTypes.Transaction, int64]
```

Comment:
does this need to use the generic interface?

seems like it could still use `keystore.Eth` and when it gets passed to the txm, then the generic interface would be used to enforce `keystore.Eth` conforms to the expected methods

Reply:
^ Yeah, I was under the impression we keep this file chain-specific because the whole file can't be generalized. It's when we pass it into the txm that the concrete type is treated as the generic interface